libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/snapshotter/model/MapboxSnapshotterOptions.kt

```diff
@@ -132,6 +144,19 @@ class MapboxSnapshotterOptions private constructor(
         fun styleUri(styleUri: String): Builder =
             apply { this.styleUri = styleUri }
 
+        /**
+         * apply routePrecision to the builder
+         * @param routePrecision String
+         * @return Builder
+         */
+        fun routePrecision(@IntRange(from = 5, to = 6)routePrecision: Int): Builder =
+            apply {
+                if (routePrecision < 5 || routePrecision > 6) throw IllegalArgumentException(
+                    "The precision values for route geometry can either be 5 or 6"
+                )
```

Comment:
```suggestion
                require(routePrecision in 5..6) {
                    "The precision values for route geometry can either be 5 or 6"
                )
```

`5` and `6` can also be extracted to `const`ants and use in the `@IntRange` and in the exception message.